diff a/board/aspeed/ast-g5/ast-g5.c b/board/aspeed/ast-g5/ast-g5.c
--- a/board/aspeed/ast-g5/ast-g5.c	2021-03-12 00:03:27.872730889 +0300
+++ b/board/aspeed/ast-g5/ast-g5.c	2021-03-12 19:25:44.213767969 +0300
@@ -58,6 +58,17 @@
 	// Set RST_BTN# (P3) to 0
 	ast_gpio_set('p', 3, 0);
 
+	int hwstrap    = (readl(AST_SCU_BASE  + 0x70) && 0x00020000) ? 1 : 0;
+	int cs         = (readl(AST_WDT2_BASE + 0x10) && 0x00000002) ? 1 : 0;
+	int wdcontinue = (getenv("wdcontinue") && getenv("wdcontinue")[0] == '1') ? 1 : 0;
+
+	if ((wdcontinue == 1) && (cs == 0) && (hwstrap == 1))
+	{
+		printf("\n\n*** NOTE: don't forget to run `firmware-updater --commit` once Linux is booted normally!\n\n");
+		/* Set WD2 timer to 300 seconds */
+		writel(300000000, AST_WDT2_BASE + 0x04);
+	}
+
 	writel(0x77744303, 0x1e78a0c4); // Set correct speed for I2C3
 	writel(0x77744303, 0x1e78a144); // Set correct speed for I2C5
 	writel(readl(0x1e6e2088) & 0xffffffb4, 0x1e6e2088); // Set TinySPI GPIO to correct mode: zero bits 0,1,2,6 of SCU88.
